lib/record_store/zone/config.rb

```diff
@@ -3,13 +3,16 @@ class Zone
     class Config
       include ActiveModel::Validations
 
-      attr_reader :ignore_patterns, :provider
+      attr_reader :ignore_patterns, :provider, :supports_alias
+      alias supports_alias? supports_alias
 
       validate :validate_zone_config
 
-      def initialize(ignore_patterns: [], provider: nil)
+      def initialize(ignore_patterns: [], provider: nil, supports_alias: nil)
         @ignore_patterns = ignore_patterns
         @provider = provider
+        @supports_alias = supports_alias
+        @supports_alias = Provider.const_get(provider).supports_alias? if @supports_alias.nil? && valid_provider?
```

Comment:
I'm okay with keeping it as is - if we move the logic into a public method, we wouldn't be able to inspect the supports_alias value after the Config object is initialized like so: https://github.com/Shopify/record_store/blob/f0f599d7e21ba2cbfd4551db135a12df9d388bf4/test/zone_test.rb#L405